picard/webservice/__init__.py

```diff
@@ -292,40 +225,46 @@ def _handle_reply(self, reply, request, handler, xml, refresh):
                         self.get(redirect_host,
                                  redirect_port,
                                  redirect_path,
-                                 handler, xml, priority=True, important=True, refresh=refresh, queryargs=redirect_query,
+                                 handler, parse_response_format, priority=True, important=True, refresh=refresh, queryargs=redirect_query,
                                  cacheloadcontrol=request.attribute(QtNetwork.QNetworkRequest.CacheLoadControlAttribute))
                     else:
                         log.error("Redirect loop: %s",
                                   reply.request().url().toString(QUrl.RemoveUserInfo)
                                   )
                         handler(reply.readAll(), reply, error)
-                elif xml:
-                    document = _read_xml(QXmlStreamReader(reply))
-                    handler(document, reply, error)
+                elif parse_response_format:
+                    response_parser = self.response_parser(parse_response_format)
+                    try:
+                        document = response_parser(reply)
+                    except Exception as e:
```

Comment:
I think we should have a specific exception class here (ExceptionResponseParserUnknown?)